ui/client/components/graph/node-modal/editors/expression/EditorType.js

```diff
@@ -13,32 +13,54 @@ class EditorType {
         return !displayRawEditor && this.switchableToBooleanEditor(expressionObj) ? Types.BOOL_EDITOR : Types.RAW_EDITOR
       case Types.BOOLEAN:
         return !displayRawEditor && this.switchableToBooleanEditor(expressionObj) ? Types.BOOL_EDITOR : Types.RAW_EDITOR
+      case Types.STRING:
+        return !displayRawEditor && this.switchableToStringEditor(expressionObj) ? Types.STRING_EDITOR : Types.RAW_EDITOR
       case Types.EXPRESSION_WITH_FIXED_VALUES || Types.RAW_EDITOR:
         return fieldType
       default:
         return Types.RAW_EDITOR
     }
   }
 
+  switchable = (editorName, expressionObj, fieldType) => {
```

Comment:
wouldn't it be more clear if custom editors were classes with switchable method, checking both type and custom conditions? And also some nonSwitchableHint field/method? Now the logic is pretty scattered and it's not so clear how to add new editors without adding many case expressions...

Reply:
I've moved some logic to editors